### PR TITLE
ID Localization for Service Topology Page

### PR DESCRIPTION
**Pull request by ariscahyadi**
Hi,

Translation has been completed to address issue #20111

### CODE REVIEW

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+{{% capture body %}}
+
+## Pengantar
+
+Secara _default_, trafik yang dikirim ke `ClusterIP` atau` NodePort` dari layanan
+dapat dialihkan ke alamat _backend_ untuk layanan tersebut. Sejak Kubernetes 1.7
+dimungkinkan untuk merutekan trafik "eksternal" ke Pod yang berjalan di 
+Node yang menerima trafik, tetapi fitur ini tidak didukung untuk `ClusterIP` dari
+layanan, dan topologi yang lebih kompleks &mdash; seperti rute zonasi &mdash;
+belum memungkinkan. Fitur topologi layanan mengatasi kekurangan ini dengan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:34]:
```suggestion
belum memungkinkan. Fitur topologi Service mengatasi kekurangan ini dengan 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+
+## Pengantar
+
+Secara _default_, trafik yang dikirim ke `ClusterIP` atau` NodePort` dari layanan
+dapat dialihkan ke alamat _backend_ untuk layanan tersebut. Sejak Kubernetes 1.7
+dimungkinkan untuk merutekan trafik "eksternal" ke Pod yang berjalan di 
+Node yang menerima trafik, tetapi fitur ini tidak didukung untuk `ClusterIP` dari
+layanan, dan topologi yang lebih kompleks &mdash; seperti rute zonasi &mdash;
+belum memungkinkan. Fitur topologi layanan mengatasi kekurangan ini dengan 
+mengizinkan pembuat layanan untuk mendefinisikan kebijakan dalam merutekan trafik
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:35]:
```suggestion
mengizinkan pembuat layanan untuk mendefinisikan kebijakan dalam merutekan lalu lintas jaringan

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+berdasarkan label Node untuk Node-Node asal dan tujuan.
+
+Dengan menggunakan label Node yang sesuai antara asal dan tujuan, operator dapat
+menunjuk kelompok Node yang "lebih dekat" dan "lebih jauh" antara satu sama lain,
+dengan menggunakan metrik apa pun yang masuk akal untuk memenuhi persyaratan 
+dari operator itu. Untuk sebagian besar operator di publik _cloud_, misalnya, ada
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:41]:
```suggestion
dari operator itu. Untuk sebagian besar operator di _cloud_ publik, misalnya, ada 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+layanan, dan topologi yang lebih kompleks &mdash; seperti rute zonasi &mdash;
+belum memungkinkan. Fitur topologi layanan mengatasi kekurangan ini dengan 
+mengizinkan pembuat layanan untuk mendefinisikan kebijakan dalam merutekan trafik
+berdasarkan label Node untuk Node-Node asal dan tujuan.
+
+Dengan menggunakan label Node yang sesuai antara asal dan tujuan, operator dapat
+menunjuk kelompok Node yang "lebih dekat" dan "lebih jauh" antara satu sama lain,
+dengan menggunakan metrik apa pun yang masuk akal untuk memenuhi persyaratan 
+dari operator itu. Untuk sebagian besar operator di publik _cloud_, misalnya, ada 
+preferensi untuk menjaga layanan trafik dalam zona yang sama, karena trafik
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:42]:
```suggestion
preferensi untuk menjaga layanan lalu lintas jaringan dalam zona yang sama, karena lalu lintas jaringan
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+antar zona memiliki biaya yang dibebankan untuk itu, sementara trafik
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:43]:
```suggestion
antar zona memiliki biaya yang dibebankan, sementara lalu lintas jaringan
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+intra zona tidak. Kebutuhan umum lainnya termasuk kemampuan untuk merutekan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:44]:
```suggestion
dalam zona yang sama tidak ada biaya. Kebutuhan umum lainnya termasuk kemampuan untuk merutekan 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+berdasarkan label Node untuk Node-Node asal dan tujuan.
+
+Dengan menggunakan label Node yang sesuai antara asal dan tujuan, operator dapat
+menunjuk kelompok Node yang "lebih dekat" dan "lebih jauh" antara satu sama lain,
+dengan menggunakan metrik apa pun yang masuk akal untuk memenuhi persyaratan 
+dari operator itu. Untuk sebagian besar operator di publik _cloud_, misalnya, ada 
+preferensi untuk menjaga layanan trafik dalam zona yang sama, karena trafik
+antar zona memiliki biaya yang dibebankan untuk itu, sementara trafik
+intra zona tidak. Kebutuhan umum lainnya termasuk kemampuan untuk merutekan 
+trafik ke Pod lokal yang dikelola oleh sebuah DaemonSet, atau menjaga traffic ke
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:45]:
```suggestion
lalu lintas jaringan ke Pod lokal yang dikelola oleh sebuah DaemonSet, atau menjaga lalu lintas jaringan ke 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+
+Dengan menggunakan label Node yang sesuai antara asal dan tujuan, operator dapat
+menunjuk kelompok Node yang "lebih dekat" dan "lebih jauh" antara satu sama lain,
+dengan menggunakan metrik apa pun yang masuk akal untuk memenuhi persyaratan 
+dari operator itu. Untuk sebagian besar operator di publik _cloud_, misalnya, ada 
+preferensi untuk menjaga layanan trafik dalam zona yang sama, karena trafik
+antar zona memiliki biaya yang dibebankan untuk itu, sementara trafik
+intra zona tidak. Kebutuhan umum lainnya termasuk kemampuan untuk merutekan 
+trafik ke Pod lokal yang dikelola oleh sebuah DaemonSet, atau menjaga traffic ke 
+Node-Node yang terhubung ke _top-of-rack switch_ yang sama untuk mendapatkan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:46]:
```suggestion
Node yang terhubung ke _top-of-rack switch_ yang sama untuk mendapatkan 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+dari operator itu. Untuk sebagian besar operator di publik _cloud_, misalnya, ada 
+preferensi untuk menjaga layanan trafik dalam zona yang sama, karena trafik
+antar zona memiliki biaya yang dibebankan untuk itu, sementara trafik
+intra zona tidak. Kebutuhan umum lainnya termasuk kemampuan untuk merutekan 
+trafik ke Pod lokal yang dikelola oleh sebuah DaemonSet, atau menjaga traffic ke 
+Node-Node yang terhubung ke _top-of-rack switch_ yang sama untuk mendapatkan 
+latensi terendah.
+
+
+## Menggunakan Topologi Layanan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:50]:
```suggestion
## Menggunakan Topologi Service
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+
+Jika klaster kamu mengaktifkan topologi layanan kamu dapat mengontrol rute trafik layanan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:52]:
```suggestion
Jika klaster kamu mengaktifkan topologi Service kamu dapat mengontrol rute lalu lintas jaringan Service
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+dengan mengatur bagian `topologyKeys` pada spesifikasi layanan. Bagian ini
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:53]:
```suggestion
dengan mengatur bagian `topologyKeys` pada spesifikasi Service. Bagian ini
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 12]
```diff
@@ -0,0 +1,206 @@
...
+adalah daftar urutan label-label Node yang akan digunakan untuk mengurutkan _endpoint_
+saat mengakses layanan ini. Trafik akan diarahkan ke Node yang nilai
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:55]:
```suggestion
saat mengakses Service ini. Lalu lintas jaringan akan diarahkan ke Node yang nilai
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 13]
```diff
@@ -0,0 +1,206 @@
...
+label pertamanya cocok dengan nilai dari Node asal untuk label yang sama. Jika
+tidak ada _backend_ untuk layanan pada Node yang sesuai, maka label kedua akan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:57]:
```suggestion
tidak ada _backend_ untuk Service pada Node yang sesuai, maka label kedua akan
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 14]
```diff
@@ -0,0 +1,206 @@
...
+dipertimbangkan, dan seterusnya, sampai tidak ada label yang tersisa.
+
+Jika tidak ditemukan kecocokan, trafik akan ditolak, sama seperti jika tidak ada
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:60]:
```suggestion
Jika tidak ditemukan kecocokan, lalu lintas jaringan akan ditolak, sama seperti jika tidak ada
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+sama sekali _backend_ untuk layanan tersebut. Artinya, _endpoint_ dipilih
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:61]:
```suggestion
sama sekali _backend_ untuk Service tersebut. Artinya, _endpoint_ dipilih 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 16]
```diff
@@ -0,0 +1,206 @@
...
+saat mengakses layanan ini. Trafik akan diarahkan ke Node yang nilai
+label pertamanya cocok dengan nilai dari Node asal untuk label yang sama. Jika
+tidak ada _backend_ untuk layanan pada Node yang sesuai, maka label kedua akan
+dipertimbangkan, dan seterusnya, sampai tidak ada label yang tersisa.
+
+Jika tidak ditemukan kecocokan, trafik akan ditolak, sama seperti jika tidak ada
+sama sekali _backend_ untuk layanan tersebut. Artinya, _endpoint_ dipilih 
+berdasarkan kunci topologi yang pertama yang tersedia pada _backend_. Jika dalam
+bagian ini ditentukan dan semua entri tidak memiliki _backend_ yang sesuai dengan
+topologi klien, layanan tidak memiliki _backend_ untuk klien dan koneksi harus
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:64]:
```suggestion
topologi klien, maka Service tidak memiliki _backend_ untuk klien dan koneksi harus 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 17]
```diff
@@ -0,0 +1,206 @@
...
+Jika tidak ditemukan kecocokan, trafik akan ditolak, sama seperti jika tidak ada
+sama sekali _backend_ untuk layanan tersebut. Artinya, _endpoint_ dipilih 
+berdasarkan kunci topologi yang pertama yang tersedia pada _backend_. Jika dalam
+bagian ini ditentukan dan semua entri tidak memiliki _backend_ yang sesuai dengan
+topologi klien, layanan tidak memiliki _backend_ untuk klien dan koneksi harus 
+digagalkan. Nilai khusus `" * "` dapat digunakan untuk mengartikan "topologi 
+apa saja". Nilai _catch-all_ ini, jika digunakan, maka hanya sebagai 
+nilai terakhir dalam daftar.
+
+Jika `topologyKeys` tidak ditentukan atau kosong, tidak ada pertimbangan topologi
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:69]:
```suggestion
Jika `topologyKeys` tidak ditentukan atau kosong, tidak ada batasan topologi 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 18]
```diff
@@ -0,0 +1,206 @@
...
+bagian ini ditentukan dan semua entri tidak memiliki _backend_ yang sesuai dengan
+topologi klien, layanan tidak memiliki _backend_ untuk klien dan koneksi harus 
+digagalkan. Nilai khusus `" * "` dapat digunakan untuk mengartikan "topologi 
+apa saja". Nilai _catch-all_ ini, jika digunakan, maka hanya sebagai 
+nilai terakhir dalam daftar.
+
+Jika `topologyKeys` tidak ditentukan atau kosong, tidak ada pertimbangan topologi 
+yang akan diterapkan.
+
+Seandainya sebuah klaster dengan Node-Node yang dilabeli dengan nama _host_ ,
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:72]:
```suggestion
Seandainya sebuah klaster dengan Node yang dilabeli dengan nama _host_ , 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 19]
```diff
@@ -0,0 +1,206 @@
...
+digagalkan. Nilai khusus `" * "` dapat digunakan untuk mengartikan "topologi 
+apa saja". Nilai _catch-all_ ini, jika digunakan, maka hanya sebagai 
+nilai terakhir dalam daftar.
+
+Jika `topologyKeys` tidak ditentukan atau kosong, tidak ada pertimbangan topologi 
+yang akan diterapkan.
+
+Seandainya sebuah klaster dengan Node-Node yang dilabeli dengan nama _host_ , 
+nama zona, dan nama wilayah mereka, maka kamu dapat mengatur nilai 
+`topologyKeys` dari sebuah layanan untuk mengarahkan trafik seperti berikut ini.
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:74]:
```suggestion
`topologyKeys` dari sebuah Service untuk mengarahkan lalu lintas jaringan seperti berikut ini.
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 20]
```diff
@@ -0,0 +1,206 @@
...
+
+* Hanya ke _endpoint_ dalam Node yang sama, gagal jika tidak ada _endpoint_ pada
+  Node : `[" kubernetes.io/hostname "]`.
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:77]:
```suggestion
  Node: `[" kubernetes.io/hostname "]`.
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 21]
```diff
@@ -0,0 +1,206 @@
...
+* Lebih menyukai ke _endpoint_ dalam Node yang sama, _fallback_ ke _endpoint_ pada
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:78]:
```suggestion
* Lebih memilih ke _endpoint_ dalam Node yang sama, jika tidak ditemukan maka ke _endpoint_ pada
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 22]
```diff
@@ -0,0 +1,206 @@
...
+  zona yang sama, diikuti oleh wilayah yang sama, dan selain itu gagal: 
+  `[" kubernetes.io/hostname ", "topology.kubernetes.io/zone", "topology.kubernetes.io/region"] `.
+  Ini mungkin berguna, misalnya, dalam kasus di mana lokalitas data sangat penting.
+* Lebih menyukai zona yang sama, tetapi _fallback_ ke _endpoint_ mana saja yang
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:82]:
```suggestion
* Lebih memilih ke _endpoint_ dalam zona yang sama, tetapi memilih _endpoint_ mana saja yang 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 23]
```diff
@@ -0,0 +1,206 @@
...
+* Lebih menyukai ke _endpoint_ dalam Node yang sama, _fallback_ ke _endpoint_ pada
+  zona yang sama, diikuti oleh wilayah yang sama, dan selain itu gagal: 
+  `[" kubernetes.io/hostname ", "topology.kubernetes.io/zone", "topology.kubernetes.io/region"] `.
+  Ini mungkin berguna, misalnya, dalam kasus di mana lokalitas data sangat penting.
+* Lebih menyukai zona yang sama, tetapi _fallback_ ke _endpoint_ mana saja yang 
+  tersedia apabila tidak ada yang tersedia dalam zona ini:
+  `[" topology.kubernetes.io/zone "," * "]`.
+
+
+## Pertimbangan-pertimbangan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:87]:
```suggestion
## Pertimbangan-pertimbangan
```
```suggestion
## Batasan
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 24]
```diff
@@ -0,0 +1,206 @@
...
+
+* Topologi layanan tidak kompatibel dengan `externalTrafficPolicy = Local`, dan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:89]:
```suggestion
* Topologi Service tidak kompatibel dengan `externalTrafficPolicy=Local`, dan
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 25]
```diff
@@ -0,0 +1,206 @@
...
+   karena itu layanan tidak dapat menggunakan kedua fitur ini sekaligus. Dimungkinkan untuk menggunakan
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:90]:
```suggestion
   karena itu Service tidak dapat menggunakan kedua fitur ini sekaligus. Dimungkinkan untuk menggunakan
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 26]
```diff
@@ -0,0 +1,206 @@
...
+   kedua fitur pada klaster yang sama untuk layanan yang berbeda, bukan untuk
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:91]:
```suggestion
   kedua fitur pada klaster yang sama untuk Service yang berbeda, bukan untuk
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 27]
```diff
@@ -0,0 +1,206 @@
...
+   layanan yang sama.
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:92]:
```suggestion
   Service yang sama.
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 28]
```diff
@@ -0,0 +1,206 @@
...
+   `topology.kubernetes.io/zone`, dan` topology.kubernetes.io/region`, tetapi akan
+   digeneralisasikan ke label Node yang lain di masa depan.
+
+* Kunci topologi harus merupakan kunci label yang valid dan paling banyak hanya 16 kunci yang dapat ditentukan.
+
+* Nilai _catch-all_, `" * "`, harus menjadi nilai terakhir pada kunci topologi, jika
+   nilai itu digunakan.
+
+
+## Contoh-contoh
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:104]:
```suggestion
## Contoh
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 29]
```diff
@@ -0,0 +1,206 @@
...
+
+Berikut ini adalah contoh umum penggunaan fitur topologi layanan.
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:106]:
```suggestion
Berikut ini adalah contoh umum penggunaan fitur topologi Service.
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 30]
```diff
@@ -0,0 +1,206 @@
...
+
+### Hanya beberapa local endpoint dari Node
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:108]:
```suggestion
### Hanya pada _endpoint_ pada Node lokal
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 31]
```diff
@@ -0,0 +1,206 @@
...
+
+Layanan yang hanya merutekan ke _endpoint_ pada lokal Node. Jika tidak ada _endpoint_ pada Node, trafik akan dihapus:
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:110]:
```suggestion
Service yang hanya merutekan ke _endpoint_ pada Node lokal. Jika tidak ada _endpoint_ pada Node, lalu lintas jaringan akan dihentikan:
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 32]
```diff
@@ -0,0 +1,206 @@
...
+    app: my-app
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  topologyKeys:
+    - "kubernetes.io/hostname"
+```
+
+### Lebih menyukai beberapa lokal _endpoint_ pada Node
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:128]:
```suggestion
### Lebih memilih _endpoint_ pada Node lokal
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 33]
```diff
@@ -0,0 +1,206 @@
...
+
+Layanan yang lebih memilih _endpoint_ pada lokal Node tetapi _fallback_ ke _endpoint_
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:130]:
```suggestion
Service yang lebih memilih _endpoint_ pada Node lokal, tetapi akan memilih ke _endpoint_ 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 34]
```diff
@@ -0,0 +1,206 @@
...
+      port: 80
+      targetPort: 9376
+  topologyKeys:
+    - "kubernetes.io/hostname"
+```
+
+### Lebih menyukai beberapa lokal _endpoint_ pada Node
+
+Layanan yang lebih memilih _endpoint_ pada lokal Node tetapi _fallback_ ke _endpoint_ 
+dalam klaster, jika _endpoint_ pada lokal Node tidak ada:
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:131]:
```suggestion
dalam klaster jika _endpoint_ pada Node lokal tidak ada:
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 35]
```diff
@@ -0,0 +1,206 @@
...
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  topologyKeys:
+    - "kubernetes.io/hostname"
+    - "*"
+```
+
+
+### Hanya untuk _Endpoint-Endpoint_ Zona atau Wilayah
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:151]:
```suggestion
### Hanya untuk _endpoint_ pada Zona atau Wilayah
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 36]
```diff
@@ -0,0 +1,206 @@
...
+
+Layanan yang lebih memilih _endpoint_ zona daripada wilayah. Jika tidak ada _endpoint_ pada
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:153]:
```suggestion
Service yang lebih memilih _endpoint_ zona daripada wilayah. Jika tidak ada _endpoint_ pada  
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 37]
```diff
@@ -0,0 +1,206 @@
...
+  topologyKeys:
+    - "kubernetes.io/hostname"
+    - "*"
+```
+
+
+### Hanya untuk _Endpoint-Endpoint_ Zona atau Wilayah
+
+Layanan yang lebih memilih _endpoint_ zona daripada wilayah. Jika tidak ada _endpoint_ pada  
+di keduanya, make trafik akan dihentikan.
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:154]:
```suggestion
keduanya, maka lalu lintas jaringan akan dihentikan.
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 38]
```diff
@@ -0,0 +1,206 @@
...
+  ports:
+    - protocol: TCP
+      port: 80
+      targetPort: 9376
+  topologyKeys:
+    - "topology.kubernetes.io/zone"
+    - "topology.kubernetes.io/region"
+```
+
+### Lebih suka Node Lokal, Zonal, kemudian Titik Akhir Regional
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:174]:
```suggestion
### Lebih memilih _endpoint_ pada Node Lokal, Zonal, terakhir Regional
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 39]
```diff
@@ -0,0 +1,206 @@
...
+
+Layanan yang lebih memilih _endpoint_ lokal, zona, kemudian _endpoint_ regional
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:176]:
```suggestion
Service yang lebih memilih _endpoint_ pada Node lokal, zonal, kemudian regional 
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 40]
```diff
@@ -0,0 +1,206 @@
...
+      targetPort: 9376
+  topologyKeys:
+    - "topology.kubernetes.io/zone"
+    - "topology.kubernetes.io/region"
+```
+
+### Lebih suka Node Lokal, Zonal, kemudian Titik Akhir Regional
+
+Layanan yang lebih memilih _endpoint_ lokal, zona, kemudian _endpoint_ regional 
+tetapi _fallback_ endpoint diseluruh klaster.
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:177]:
```suggestion
tetapi jika tetap tidak ditemukan maka akan memilih  _endpoint_ diseluruh klaster.
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 41]
```diff
@@ -0,0 +1,206 @@
...
+    - "topology.kubernetes.io/region"
+    - "*"
+```
+
+
+{{% /capture %}}
+
+{{% capture whatsnext %}}
+
+* Baca tentang [mengaktifkan topologi layanan](/docs/tasks/administer-cluster/enabling-service-topology)
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:203]:
```suggestion
* Baca tentang [mengaktifkan topologi Service](/docs/tasks/administer-cluster/enabling-service-topology)
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 42]
```diff
@@ -0,0 +1,206 @@
...
+* Baca [menghubungkan aplikasi dengan layanan](/docs/concepts/services-networking/connect-applications-service/)
```
Comment (wahyuoi) [inline content/id/docs/concepts/services-networking/service-topology.md:204]:
```suggestion
* Baca [menghubungkan aplikasi dengan Service](/docs/concepts/services-networking/connect-applications-service/)
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 43]
```diff
@@ -0,0 +1,206 @@
+---
+title: Topologi Service (_Service Topology_)
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/service-topology.md:2]: As per https://github.com/kubernetes/website/issues/20112, the title doesn't work with italic format, you can fix this as follow up PR
```suggestion
title: Topologi Service (Service Topology)
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 44]
```diff
@@ -0,0 +1,206 @@
...
+feature:
+  title: Topologi Service (_Service Topology_)
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/service-topology.md:4]: As per https://github.com/kubernetes/website/issues/20112, the title doesn't work with italic format, you can fix this as follow up PR
```suggestion
title: Topologi Service (Service Topology)
```

File: content/id/docs/concepts/services-networking/service-topology.md [hunk 45]
```diff
@@ -0,0 +1,206 @@
...
+  Ini mungkin berguna, misalnya, dalam kasus di mana lokalitas data sangat penting.
+* Lebih memilih ke _endpoint_ dalam zona yang sama, tetapi memilih _endpoint_ mana saja yang 
+  tersedia apabila tidak ada yang tersedia dalam zona ini:
+  `[" topology.kubernetes.io/zone "," * "]`.
+
+
+## Batasan
+
+* Topologi Service tidak kompatibel dengan `externalTrafficPolicy=Local`, dan
+   karena itu Service tidak dapat menggunakan kedua fitur ini sekaligus. Dimungkinkan untuk menggunakan
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/service-topology.md:90]: nit: this adds extra spaces to the page, you can take a look https://deploy-preview-20229--kubernetes-io-master-staging.netlify.app/id/docs/concepts/services-networking/service-topology/#batasan

same with the lines below

Comment (girikuncoro) [reply content/id/docs/concepts/services-networking/service-topology.md:90]: you can address this as follow up PR

**Comment by girikuncoro** [comment]
Looking good, thanks for the contribution @ariscahyadi ! 🎉 I have added some minor comments, you can address those as follow-up/separate PR since it's not a major blocker. This Service Topology mechanism is very interesting.

/lgtm
/approve

File: content/id/docs/concepts/services-networking/service-topology.md (hunk 44 above)
Comment (ariscahyadi) [reply content/id/docs/concepts/services-networking/service-topology.md:4]: Ok... I will do it :)

File: content/id/docs/concepts/services-networking/service-topology.md (hunk 45 above)
Comment (ariscahyadi) [reply content/id/docs/concepts/services-networking/service-topology.md:90]: Ok... I will do it :)

**Comment by ariscahyadi** [comment]
> Looking good, thanks for the contribution @ariscahyadi ! 🎉 I have added some minor comments, you can address those as follow-up/separate PR since it's not a major blocker. This Service Topology mechanism is very interesting.
> 
> /lgtm
> /approve

You're welcome @girikuncoro ... definitely, I would love to create another PR for this :)

File: content/id/docs/concepts/services-networking/service-topology.md (hunk 45 above)
Comment (ariscahyadi) [reply content/id/docs/concepts/services-networking/service-topology.md:90]: Upload in PR #20500

**Comment by ariscahyadi** [comment]
> > Looking good, thanks for the contribution @ariscahyadi ! 🎉 I have added some minor comments, you can address those as follow-up/separate PR since it's not a major blocker. This Service Topology mechanism is very interesting.
> > /lgtm
> > /approve
> 
> You're welcome @girikuncoro ... definitely, I would love to create another PR for this :)

Upload as PR #20500

**Pull request merged**